File: 15-hacktiv/2-roman-numeral/solution.py
def toRoman(value):
  roman = {
    1000: 'M',
    500: 'D',
    100: 'C',
    50: 'L',
    10: 'X',
    5: 'V',
    1: 'I',
  }
  result = ''
  for key in roman:
    while value >= key:
      result += roman[key]
      value -= key
  return result

print('My totally sweet testing script\n')
print('input | expected | actual')
print('------------|--------------------|----------------')
# print('4    | IV      | ', toRoman(4))
# print('9    | IX      | ', toRoman(9))
# print('23   | XXIII   | ', toRoman(23))
# print('1453 | MCDLIII | ', toRoman(1453))
# print('1646 | MDCXLVI | ', toRoman(1646))

File: 15-hacktiv/2-roman-numeral/solution.go
package main

import "fmt"

func toRoman(value int) string {
	number := []int{1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1}
	roman := []string{"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"}

	result := ""
	for i := 0; i < len(number); i++ {
		for value >= number[i] {
			result += roman[i]
			value -= number[i]
		}
	}
	return result
}

func main() {
	fmt.Println("My totally sweet testing script")
	fmt.Println("input | expected | actual")
	fmt.Println("------|----------|--------")
	fmt.Println("4     | IV       | ", toRoman(4))
	fmt.Println("9     | IX       | ", toRoman(9))
	fmt.Println("23    | XXIII    | ", toRoman(23))
	fmt.Println("1453  | MCDLIII  | ", toRoman(1453))
	fmt.Println("1646  | MDCXLVI  | ", toRoman(1646))
}
